src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AttendancesService.kt

```
@@ -37,38 +36,46 @@ class AttendancesService(
 
   @Transactional
   fun mark(principalName: String, attendances: List<AttendanceUpdateRequest>) {
-    val attendanceUpdatesById = attendances.associateBy { it.id }
-    val attendanceReasonsByCode = attendanceReasonRepository.findAll().associateBy { it.code.toString() }
+    val attendanceUpdatesById = attendances.onEach(AttendanceUpdateRequestValidator::validate).associateBy { it.id }
```

We should be able to determine from the update requests if they are valid.

Intention is to validate each request and fail fast on the first failure if there is one.